topics-zh-CN/concurrent-mutability.md

```diff
@@ -136,71 +109,41 @@ object MyListCache {
 }
 ```
 
-You will need to implement some form of locking or check-and-set logic to ensure proper concurrency.
-
 ## Thread-isolated state
 
-[Rule 1 of Kotlin/Native state](concurrency-overview.md#rule-1-mutable-state-1-thread) is that a mutable state is 
-visible to only one thread. Atomics allow mutability from any thread. 
-Isolating a mutable state to a single thread, and allowing other threads to communicate with that state, is an alternative 
-method for achieving concurrent mutability.
-
-To do this, create a work queue that has exclusive access to a thread, and create a mutable state that 
-lives in just that thread. Other threads communicate with the mutable thread by scheduling _work_ on the work queue.
-
-Data that goes in or comes out, if any, needs to be frozen, but the mutable state hidden in the worker thread remains 
-mutable. 
+正如规则一揭示的那样，Kotlin/Native要求一个可变状态只能对一个线程可见。通过Atomics支持，可变状态可以被多线程访问。另外一种解决思路是：将可变状态隔离到单个线程，其余线程通过线程间通信操作获取该状态。
 
-Conceptually it looks like the following: one thread pushes a frozen state into the state worker, which stores it in 
-the mutable state container. Another thread later schedules work that takes that state out.
+线程隔离状态的设计通常是创建一个可供其他线程进行排他访问持有可变状态线程的工作队列，通过该工作队列调度即可完成对线程隔离可变状态的跨线程访问。当然从该工作队列写入和读取的状态副本是冻结的，而线程内的可变状态可根据情况返回其副本或者使用外部冻结状态进行更新。简单归纳：一个线程将冻结状态通过工作队列传递给目标线程，由其更新内部持有的可变状态，随后其他线程通过该工作队列获取该状态副本。
 
 ![Thread-isolated state](isolated-state.png){animated="true"}
 
-Implementing thread-isolated states is somewhat complex, but there are libraries that provide this functionality.
+实现线程隔离状态有些复杂，但有提供此功能的库。
 
 ### `AtomicReference` vs. thread-isolated state
 
-For simple values, `AtomicReference` will likely be an easier option. For cases with significant states, and potentially 
-significant state changes, using a thread-isolated state may be a better choice. The main performance penalty is actually crossing 
-over threads. But in performance tests with collections, for example, a thread-isolated state significantly outperforms a
-mutable state implemented with `AtomicReference`.
+对于简单的值， `AtomicReference` 可能是一个不错的选择；但是对于那些频繁变动的复杂的状态，使用线程隔离的状态可能是一个更好的选，虽然跨线程调用通常被认为会带来更大的性能损耗，但是大量性能测试数据表明很多情况下线程隔离状态优于 `AtomicReference` ，比如`collections` 。
 
-The thread-isolated state also avoids the consistency issues that `AtomicReference` has. Because all operations happen in the 
-state thread, and because you're scheduling work, you can perform operations with multiple steps and guarantee consistency 
-without managing thread exclusion. Thread isolation is a design feature of the Kotlin/Native state rules, and 
-isolating mutable states works with those rules.
+此外，使用线程隔离状态可以避免 `AtomicReference` 带来的一致性问题。因为所有的代码指令都是在工作线程内完成，也即所有的操作都是单线程执行的，此时不需要进行大量并发保护来保证一致性。线程隔离的状态是Kotlin/Native的一个设计特性，可以很好的满足状态共享和可变状态的规范。
```

Comment:
state thread -> 状态线程 （工作线程不准确）
 and because you're scheduling work, you can perform operations with multiple steps and guarantee consistency 
 without managing thread exclusion. 这句漏了。